### ISSUE TITLE
Simplified `docker compose` deployment

### ISSUE THREAD

**Issue by ClementNerma** [bigcapitalhq/bigcapital#706]
Hi there!

I've been very interested to use BigCapital, but I'm finding the `docker compose` deployment process to be too complex.

Basically, you have to clone a 60MB repository with quite a few files, use an alternative `docker compose -f ...` command to use the production file specifically, and it's "hardcoded" into the repository.

For instance, I want the volumes to point to a specific folder instead of being Docker named volumes. If I modify the production compose file, it will be overriden the next time I `git pull` this project.

Also, updating the install requires both to pull the repository and _then_ to update the containers.

Would it be possible to have a simplified `docker-compose.yml` file which contains everything needed **without any other dependency**, i.e. without needing to clone a repository. This is what most self-hosted projects do and it's a lot easier to maintain in the long run.

**Comment by mpkossen** [comment]
I've been looking into this, and for the most part it should be fairly simply. However you can't avoid having to create some configuration files with the way things are currently set up. The only thing that seems to stand in the way is the migration container, which must be built whenever there's an update. It would be considerably easier if the webapp container handles database initialization and migration (that would apply also for BigCapital's production deployment).

Envoy should be optional for deployments, because most people won't need load balancing and have their own reverse proxy in place. If you leave that out, you could just have your proxy direct traffic to ports 80 and 3000 (for `/api`).

I'm also curious why MySQL has a `my.cnf` that binds it to 0.0.0.0, because the standard MySQL docker image listens on port 3306 by default without the need for additional configuration.

Finally, I don't think most deployments would need a customized redis configuration.

In summary, to achieve a simplified docker compose:
1. Leave out Envoy and discard the `my.cnf` and `redis.conf`.
2. Move database initialization and migration to app image.

**Comment by teloryfrozy** [comment]
@ClementNerma 
Hello,
First things first, I totally agree with @mpkossen, there should be a light version of a docker-compose file without the reverse proxy and the load balancer.
I wanted to deploy the software with [Dokploy](https://dokploy.com/), a docker wrapper with some nice UI.
Because I was unable to deploy it following the docs, I did some research and came across a [db_migration container](https://hub.docker.com/r/elestio/bigcapital-server).
For anyone trying to deploy BigCapital within a production environment; I ended up with this compose file

```yaml
services:
  webapp:
    image: bigcapitalhq/webapp:latest
    restart: unless-stopped
    depends_on:
      - server

  server:
    image: bigcapitalhq/server:latest
    restart: unless-stopped
    depends_on:
      - mysql
      - redis

    environment:
      # Mail
      - MAIL_HOST=${MAIL_HOST}
      - MAIL_USERNAME=${MAIL_USERNAME}
      - MAIL_PASSWORD=${MAIL_PASSWORD}
      - MAIL_PORT=${MAIL_PORT}
      - MAIL_SECURE=${MAIL_SECURE}
      - MAIL_FROM_NAME=${MAIL_FROM_NAME}
      - MAIL_FROM_ADDRESS=${MAIL_FROM_ADDRESS}
      # Database
      - DB_HOST=${DB_HOST}
      - DB_USER=${DB_USER}
      - DB_PASSWORD=${DB_PASSWORD}
      - DB_CHARSET=${DB_CHARSET}
      # System database
      - SYSTEM_DB_NAME=${SYSTEM_DB_NAME}
      # Tenants databases
      - TENANT_DB_NAME_PERFIX=${TENANT_DB_NAME_PERFIX}
      # Authentication
      - JWT_SECRET=${JWT_SECRET}
      # Application
      - BASE_URL=${BASE_URL}
      # Agendash
      - AGENDASH_AUTH_USER=${AGENDASH_AUTH_USER}
      - AGENDASH_AUTH_PASSWORD=${AGENDASH_AUTH_PASSWORD}
      # Sign-up restrictions
      - SIGNUP_DISABLED=${SIGNUP_DISABLED}
      - SIGNUP_ALLOWED_DOMAINS=${SIGNUP_ALLOWED_DOMAINS}
      - SIGNUP_ALLOWED_EMAILS=${SIGNUP_ALLOWED_EMAILS}
      # Sign-up email confirmation
      - SIGNUP_EMAIL_CONFIRMATION=${SIGNUP_EMAIL_CONFIRMATION}
      # Gotenberg (Pdf generator)
      - GOTENBERG_URL=${GOTENBERG_URL}
      - GOTENBERG_DOCS_URL=${GOTENBERG_DOCS_URL}
      # Exchange Rate
      - EXCHANGE_RATE_SERVICE=${EXCHANGE_RATE_SERVICE}
      - OPEN_EXCHANGE_RATE_APP_ID=${OPEN_EXCHANGE_RATE_APP_ID}
      # Bank Sync
      - BANKING_CONNECT=${BANKING_CONNECT}
      # Plaid
      - PLAID_ENV=${PLAID_ENV}
      - PLAID_CLIENT_ID=${PLAID_CLIENT_ID}
      - PLAID_SECRET=${PLAID_SECRET}
      - PLAID_LINK_WEBHOOK=${PLAID_LINK_WEBHOOK}
      # Lemon Squeez
      - LEMONSQUEEZY_API_KEY=${LEMONSQUEEZY_API_KEY}
      - LEMONSQUEEZY_STORE_ID=${LEMONSQUEEZY_STORE_ID}
      - LEMONSQUEEZY_WEBHOOK_SECRET=${LEMONSQUEEZY_WEBHOOK_SECRET}
      - HOSTED_ON_BIGCAPITAL_CLOUD=${HOSTED_ON_BIGCAPITAL_CLOUD}
      # New Relic metrics tracking
      - NEW_RELIC_DISTRIBUTED_TRACING_ENABLED=${NEW_RELIC_DISTRIBUTED_TRACING_ENABLED}
      - NEW_RELIC_LOG=${NEW_RELIC_LOG}
      - NEW_RELIC_AI_MONITORING_ENABLED=${NEW_RELIC_AI_MONITORING_ENABLED}
      - NEW_RELIC_CUSTOM_INSIGHTS_EVENTS_MAX_SAMPLES_STORED=${NEW_RELIC_CUSTOM_INSIGHTS_EVENTS_MAX_SAMPLES_STORED}
      - NEW_RELIC_SPAN_EVENTS_MAX_SAMPLES_STORED=${NEW_RELIC_SPAN_EVENTS_MAX_SAMPLES_STORED}
      - NEW_RELIC_LICENSE_KEY=${NEW_RELIC_LICENSE_KEY}
      - NEW_RELIC_APP_NAME=${NEW_RELIC_APP_NAME}
      # S3
      - S3_REGION=${S3_REGION}
      - S3_ACCESS_KEY_ID=${S3_ACCESS_KEY_ID}
      - S3_SECRET_ACCESS_KEY=${S3_SECRET_ACCESS_KEY}
      - S3_ENDPOINT=${S3_ENDPOINT}
      - S3_BUCKET=${S3_BUCKET}
      # Redis
      - REDIS_HOST=redis
      - REDIS_PORT=6379
      - QUEUE_HOST=redis
      - QUEUE_PORT=6379

  mysql:
    image: elestio/bigcapital-mariadb:latest
    restart: unless-stopped
    environment:
      - MARIADB_ROOT_PASSWORD=${DB_ROOT_PASSWORD}
      - MARIADB_DATABASE=${SYSTEM_DB_NAME}
      - MARIADB_USER=${DB_USER}
      - MARIADB_PASSWORD=${DB_PASSWORD}
    volumes:
      - mariadb:/var/lib/mariadb


  database_migration:
      image: elestio/bigcapital-migration:latest
      restart: no
      environment:
        - DB_HOST=${DB_HOST}
        - DB_USER=${DB_USER}
        - DB_PASSWORD=${DB_PASSWORD}
        - DB_CHARSET=${DB_CHARSET}
        - SYSTEM_DB_NAME=${SYSTEM_DB_NAME}
      depends_on:
        - mysql
  
  redis:
    image: redis:7-alpine
    restart: unless-stopped
    volumes:
      - redis:/data

    healthcheck:
      test: ["CMD", "redis-cli", "ping"]
      interval: 10s
      timeout: 5s
      retries: 5

  gotenberg:
    image: gotenberg/gotenberg:7
    restart: unless-stopped


volumes:
  mariadb:
    name: bigcapital_mariadb
    driver: local
  redis:
    name: bigcapital_redis
    driver: local
```
